Maths/QuadraticRoots.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @see https://www.cuemath.com/algebra/roots-of-quadratic-equation/
+ * @author Dibya Debayan Dash
+ * Calculates the roots of a quadratic equation of the form ax^2 + bx + c = 0.
+ *
+ * @param {number} a - Coefficient of x^2.
+ * @param {number} b - Coefficient of x.
+ * @param {number} c - Constant term.
+ * @returns {number[]} - An array containing the roots if they are real,
+ *                      or an empty array indicating no real roots.
+ *
+ * @example
+ * // Find the roots of the quadratic equation: 2x^2 - 4x + 2 = 0
+ * const roots = quadraticRoots(2, -4, 2);
+ * // Expected output: [1, 1]
+ */
+const quadraticRoots = (a, b, c) => {
+  // Calculate the discriminant
+  const discriminant = b * b - 4 * a * c
+
+  // Check if roots are real
+  if (discriminant < 0) {
```

Comment:
Small side note: The `else`s here are redundant due to the `return`s. Whether to keep them is a matter of taste; I don't mind either way. You could rewrite this to

```javascript
if (discriminant < 0) {
  return []
}
if (discriminant === 0) {
  return [-b / (2 * a)]
}
const sqrtDiscriminant = Math.sqrt(discriminant)
return [
  (-b + sqrtDiscriminant) / (2 * a),
  (-b - sqrtDiscriminant) / (2 * a)
]
```

Reply:
Is there any particular reason why you want to  have multiple if's instead of using if else condition "The else's here are redundant due to the returns" I can say we are using unecessay if or is it just a good practice let me know

Reply:
Should I rewrite this as well

Reply:
The `else`s normally have an impact on control flow. Here they don't, because the `return`s will already exit the function. Thus the `else`s can be omitted; if you didn't care about formatting, you could straight up delete the `else` tokens, and everything works as expected.

> I can say we are using unecessay if

No, the `if`s are not unnecessary, because they can not be removed without altering the semantics of the code.

> is it just a good practice let me know

It is ultimately a matter of taste; I have no hard feelings either way, though I prefer omitting redundant `else`s (even if this comes at the "cost" of +1 line given our current formatting guideline; I also vaguely remember a "precedent" for that in this repo). I mostly wanted to make you aware of this. Many argue that the case analysis is more readable using `if`-`else`. As said, I've approved this, this is good to go as-is, you don't need to change it.